waku/node/waku_node.nim

```diff
@@ -1197,6 +1197,17 @@ proc fetchPeerExchangePeers*(
       error = pxPeersRes.error
     return err(pxPeersRes.error)
 
+proc peerExchangeLoop(node: WakuNode) {.async.} =
+  while node.started:
+    (await node.fetchPeerExchangePeers()).isOkOr:
+      warn "error while fetching peers from peer exchange", error = error
+    await sleepAsync(1.minutes)
+
+proc startPeerExchangeLoop*(node: WakuNode) =
+  info "starting peer exchange loop"
+
+  asyncSpawn node.peerExchangeLoop()
```

Comment:
Oh makes sense! Updated it now, lmk what you think :))

With the previous approach, it would still get a graceful shutdown when the node stops, as it is inside a `while node.started:` loop.

However, with this approach we can also stop it a will and not only when the node stops, so it is definitely more robust :))

Thank you!